nixos/modules/programs/steam.nix

```diff
@@ -177,7 +177,21 @@ in {
       };
     };
 
-    programs.steam.extraPackages = config.fonts.packages;
+    programs.steam.extraPackages = config.fonts.packages ++ (
```

Comment:
This seems extremely jank. Are you sure we need all of those? And why?

Reply:
We don't magically know which part of packages in `fonts.packages`  in user's own NixOS config could fix CJK problem. The best solution to solve it is to provide all users' fonts packages to steam The vast majority of users should be satisfied to have this issue fixed, and the others who have their own ideas about CJK donts in Steam could override the default value.

Packages in `fonts.packages` should be always on the disk, so it won't waste any disk space.

Reply:
I mean the gamescope bit. I understand the fonts.

Reply:
> I mean the gamescope bit. I understand the fonts.

This part was come up with in @BillHuang2001 https://github.com/NixOS/nixpkgs/issues/162562#issuecomment-1229444338 , and do solve our problems. I don't know why and whether it is the minium, orz.

Reply:
> That comment is nearly two years old. Are you sure this is still necessary? I really don't want to have code cargo-culted around without fully understanding why we're doing things.

Yes, I'm sure. I met the same problem for gamescope this year, and fixed it as this. The issue #214275 is still open.

Reply:
That is an entirely different issue, and it should have been fixed by the buildFHSEnv changes a long time ago. Can you find or file a new issue for whatever gamescope behavior you're seeing?

Reply:
> That comment is nearly two years old. Are you sure this is still necessary? I really don't want to have code cargo-culted around without fully understanding why we're doing things.

That comment may be old but the solution is still accurate. Gamescope is not working without those tweaks.

Reply:
Even if it is, gating this on gamescopeSession is incorrect, because that starts Steam inside Gamescope, not Gamescope inside Steam. I'd also much prefer to see a solution that doesn't involve adding random things into the fhsenv.

Reply:
At the very least, the condition to activate it is wrong.

Reply:
> At the very least, the condition to activate it is wrong.

Ok I understand. It's my fault.

So which condition would you suggest. Would you think `programs.gamescope.enable` acceptable. Or we can make a new option `programs.steam.gamescopeEnhancement` or sth.

Reply:
> I'm pretty sure I have a better fix for both of your issues.

So what's it.

Reply:
I'm testing it now and will submit a PR if it works.

Reply:
`libkrb5` and `keyutils` are the only packages needed here, btw.